javascript/ql/src/semmle/javascript/security/dataflow/DOM.qll

```diff
@@ -81,9 +81,15 @@ class DomMethodCallExpr extends MethodCallExpr {
       or
       name = "appendChild" and argPos = 0
       or
-      name = "setAttribute" and argPos = 1
-      or
-      name = "setAttributeNS" and argPos = 2
+      (
+        name = "setAttribute" and argPos = 1
+        or
+        name = "setAttributeNS" and argPos = 2
+      ) and
+      // restrict to potentially dangerous attributes
+      exists(string attr | attr = "href" or attr = "src" |
+        getArgument(argPos-1).getStringValue().toLowerCase() = attr
```

Comment:
We already have `isUrlValuedAttribute` which seems to do this more generally, although it currently only works if the element type is known. Maybe we could reuse that?

Reply:
Hm, interesting thought. That predicate does rely on knowing what the element type is, though, which in this case we don't. (I'll definitely add `action` and `formaction` here, though.)